buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -397,17 +395,90 @@ private void waitForProcessToExit(ProcessHandle processHandle) {
     }
 
     private void createWorkingDir(Path distroExtractDir) throws IOException {
-        services.sync(spec -> {
-            spec.from(distroExtractDir.toFile());
-            spec.into(workingDir.toFile());
-        });
+        syncWithLinks(distroExtractDir, workingDir);
         Files.createDirectories(configFile.getParent());
         Files.createDirectories(confPathRepo);
         Files.createDirectories(confPathData);
         Files.createDirectories(confPathLogs);
         Files.createDirectories(tmpDir);
     }
 
+    /**
+     * Does the equivalent of `cp -lr` and `chmod -r a-w` to save space and improve speed.
+     * We remove write permissions to make sure files are note mistakenly edited ( e.x. the config file ) and changes
+     * reflected across all copies. Permissions are retained to be able to replace the links.
+     *
+     * @param sourceRoot where to copy from
+     * @param destinationRoot destination to link to
+     */
+    private void syncWithLinks(Path sourceRoot, Path destinationRoot) {
+        // There's some latency in Windows between when a cluster running here previously releases the files so we can
+        // can clean them up. Make sure we can run the same clusters in quick succession.
+        if (Files.exists(destinationRoot)) {
+            removeWithRetry(destinationRoot);
```

Comment:
I don't think we should have special windows logic that is always run. Linux should not need this leniency right? Additionally, when would we be in a situation where a single build is reusing the same destination dir? Any test clusters should each have their own unique dirs, so I don't see the situation that gets us needing to wait/retry on the destination being removed async.

Reply:
This is reproduced in integration tests, where the same clusters are stopped and started multiple times in quick succession. It is a possible use-case, even trough we don't use it in the build. We could set the tests up to do this cleanup and account for it, but just because we don't have a use-case to reuse a cluster directory Today, it doesn't mean we will never have one 
Do you think there's ways in which this  will hurt us ? 

I was on the fence w.r.t running this on Linux, I have a slight preference to always run platform specific logic when possible just because windows gets less testing.